src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using static StackExchange.Redis.ConnectionMultiplexer;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Azure node maintenance event. For more information, please see: https://github.com/Azure/AzureCacheForRedis/blob/main/AzureRedisEvents.md
+    /// </summary>
+    public class AzureMaintenanceEvent
+    {
+        private const string PubSubChannelName = "AzureRedisEvents";
+
+        internal AzureMaintenanceEvent(string azureEvent)
+        {
+            if (azureEvent == null)
+            {
+                return;
+            }
+
+            // The message consists of key-value pairs delimted by pipes. For example, a message might look like:
+            // NotificationType|NodeMaintenanceStarting|StartTimeUtc|2021-09-23T12:34:19|IsReplica|False|IpAddress|13.67.42.199|SSLPort|15001|NonSSLPort|13001
+            var message = new ReadOnlySpan<char>(azureEvent.ToCharArray());
```

Comment:
suggest: `var message = azureEvent.AsSpan();` - the `.ToCharArray()` forces an extra allocation; `.AsSpan()` creates the ROS directly over the existing `string`

Reply:
Thanks Marc! I haven't worked with spans too much so the pointers are appreciated :)